internal/key/age/age.go

```diff
@@ -174,3 +175,22 @@ func askPasswordStdin(r io.Reader, logger log.Logger) func() ([]byte, error) {
 		return p, nil
 	}
 }
+
+func parseAgeIdentityFunc(d string) (age.Identity, error) {
+	lines := strings.Split(d, "\n")
+
+	var privateKey string = ""
+
+	for _, line := range lines {
+		if strings.HasPrefix(line, "AGE-SECRET-KEY-1") {
+			privateKey = strings.Trim(line, " ")
+			break
+		}
+	}
+
+	if privateKey == "" {
+		return nil, fmt.Errorf("no private key found")
+	}
+
+	return age.ParseX25519Identity(privateKey)
+}
```

Comment:
What do you think if we simplify this by using regexes and `Trimspace`? Something similar to this:

```go
var agePrivKeyRegex = regexp.MustCompile("(?m)(^AGE-SECRET-KEY-.*$)")

func parseAgeIdentityFunc() func(s string) (age.Identity, error) {
	return func(d string) (age.Identity, error) {
		d = agePrivKeyRegex.FindString(d)
		d = strings.TrimSpace(d)

		return age.ParseX25519Identity(d)
	}
}

```